src/executors/DaskExecutor.py

```diff
@@ -1,23 +1,238 @@
+            )
+            futures.append(new_future)
+        return futures
+
+    def start_runs(self):
+        sampler_interface: S = self.sampler.sampler_interface
+        print(100 * "=")
+        print("Creating initial runs")
+
+        initial_parameters = self.sampler.get_initial_parameters()
+        futures = self.submit_batch_of_params(initial_parameters)
+        completed = 0
+
+        if sampler_interface in [S.SEQUENTIAL]:
+            seq = as_completed(futures)
+            for future in seq:
+                res = future.result()
+                completed += 1
+                if self.sampler.total_budget > completed:
+                    params = self.sampler.get_next_parameter()
+                    new_future = self.simulator_client.submit(
+                        run_simulation_task, self.runner_args, params, self.base_run_dir
+                    )
+                    seq.add(new_future)
+
+        elif sampler_interface in [S.BATCH]:
+            while completed < self.sampler.total_budget:
+                seq = wait(futures)
+                param_list = self.sampler.get_next_parameter()
+                futures = self.submit_batch_of_params(param_list)
+                completed += len(futures)
+                print("BATCH SAMPLER", 20 * "=", completed, 20 * "=")
+
+        elif sampler_interface in [S.ACTIVE, S.ACTIVEDB]:
+            iterations = 0
+            while True:
+                print(
+                    20 * "=",
+                    f"Iteration {iterations}; ",
+                    f"samples collected: {completed}",
+                    20 * "=",
+                )
+                print(self.sampler.parser.print_dset_sizes())
+                # NOTE: ------ Run Samples and block until completed ------
+                seq = wait(
+                    futures
+                )  # outputs should be a list of tuples we ignore in this case
+
+                # NOTE: ------ Collect outputs ------
+                outputs = []
+                for res in seq.done:
+                    outputs.append(res.result())
+                outputs = self.sampler.collect_batch_results(outputs)
+
+                # NOTE: ------ Check if out of budget ------
+                completed += len(outputs)
+                if completed > self.sampler.total_budget:
+                    break
+
+                # NOTE: ------ Update the pool and training data from outputs ------
+                self.sampler.parser.update_pool_and_train(outputs)
+
+                # print('Updated Pool size: ', len(self.sampler.parser.pool), 'Train size: ', len(self.sampler.parser.train))
+                # print(self.sampler.parser.return_dset_sizes())
+
+                # NOTE: Collect next data for training
+
+                train, valid, test, pool = (
+                    self.sampler.parser.get_train_valid_test_pool()
+                )
+                # rescale data and pool
+                train, valid, test, pool = (
```

Comment:
maybe better to call the variables `train_scaled`, `valid_scaled` and so on. Sorry for being so pedantic with this but I think this back and forth between scaled and unscaled variables could get confusing

Reply:
Was wanting to keep them the same name as to reduce memory usage...